### PR TITLE
posix: check transaction stage in more places

### PR DESCRIPTION
**Pull request by marcinslusarz**
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemfile/156)

### CODE REVIEW

**Comment by krzycz** [comment]
:lgtm:

---

Reviewed 25 of 25 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/156#-:-KlKZCv_cX6xdkxBn3tJ:bnfp4nl)*

File: src/libpmemfile-posix/utils.h [hunk 1]
```diff
@@ -36,6 +36,9 @@
 #include "inode.h"
 #include "layout.h"
 
+#define ASSERT_IN_TX() ASSERTeq(pmemobj_tx_stage(), TX_STAGE_WORK)
```
Comment (GBuella) [inline src/libpmemfile-posix/utils.h]: Could these two be functions instead of macros? I can live with this, just asking...

Comment (marcinslusarz) [inline src/libpmemfile-posix/utils.h]: Then we would lose file/line information.

**Comment by GBuella** [comment]
:lgtm:

---

Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/156#-:-KlTIWnPhpPDhm5f6e7Z:bnfp4nl)*

**Pull request merged**